app/models/trainee.rb

```diff
@@ -209,7 +209,7 @@ def trn_received!(new_trn = nil)
 
     # Skip deferred and withdrawn to avoid state change
     # but to still register trn
-    receive_trn! unless deferred? || withdrawn?
+    receive_trn! unless deferred? || withdrawn? || trn_received?
```

Comment:
Is this to make it idempotent?

I feels like it more

```
    event :receive_trn do
      transition %i[submitted_for_trn deferred trn_received] => :trn_received
    end
```

Reply:
Happy to change. excuse my pre-coffee skillz